### DISCUSSION TITLE
How to pass '--first-parent' to Repo.iter_commits() ?

### DISCUSSION THREAD

**Discussion by mbanders** [gitpython-developers/GitPython#1357] [Q&A] [answered]
The command `git rev-list` has a lot of options including a very useful one `--first-parent`: https://git-scm.com/docs/git-rev-list

Is there a way to pass '--first-parent' as an option to `Repo.iter_commits()` ?

**Comment by Byron** [comment, marked as answer]
As the documentation of [`iter_commts()`](https://gitpython.readthedocs.io/en/stable/reference.html?highlight=iter_commits#git.repo.base.Repo.iter_commits) suggests keyword arguments can be used to pass additional arguments to the underlying git command. Specifically, `repo.iter_commits(first_parent=true)` should do the job.

Passing arguments works in most of the GitPython API, [here is more on how parameters are converted to CLI flags](https://gitpython.readthedocs.io/en/stable/tutorial.html#using-git-directly).
